transform/mattermost-analytics/models/marts/web_app/platform/_platform__models.yml

```diff
@@ -0,0 +1,90 @@
+      - name: user_actual_role               
+      - name: context_page_url               
+      - name: context_os_name                
+      - name: context_page_title             
+      - name: anonymous_id                   
+      - name: context_app_build              
+      - name: context_page_search            
+      - name: context_library_version        
+      - name: context_ip                     
+      - name: context_useragent              
+      - name: context_app_name               
+      - name: context_locale                 
+      - name: context_screen_density         
+      - name: context_page_path              
+      - name: context_os_version             
+      - name: category                       
+      - name: duration                     
+      - name: num_of_request                 
+      - name: max_api_resource_size          
+      - name: longest_api_resource_duration  
+      - name: user_id                        
+      - name: userid                         
+      - name: root_id                        
+      - name: post_id                        
+      - name: sort                           
+      - name: team_id                        
+      - name: version                        
+      - name: keyword                        
+      - name: count                          
+      - name: gfyid                          
+      - name: field                          
+      - name: plugin_id                      
+      - name: installed_version              
+      - name: group_constrained              
+      - name: value                          
+      - name: include_deleted                
+      - name: role                           
+      - name: privacy                        
+      - name: scheme_id                      
+      - name: metric                         
+      - name: error                          
+      - name: num_invitations_sent           
+      - name: num_invitations                
+      - name: channel_sidebar                
+      - name: app                            
+      - name: method                         
+      - name: remaining                      
+      - name: screen                         
+      - name: filter                         
+      - name: version                        
+      - name: original_timestamp             
+      - name: sent_at                        
+      - name: uuid_ts                        
+      - name: received_at
+      - name: event_date
+        tests:
+          - not_null 
+      - name: received_at_date
+
+    tests:
+      # Each event should appear once
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - timestamp
+            - received_at
+            - original_timestamp
+            - user_id
+            - user_actual_id
+            - category
+            - type
+            - duration
+            - id
```

Comment:
this test, unique and not null test on `id` is failing, I tried this initially

Reply:
I have tried multiple variations of unique combination of test , finally was able to crack it. The individual tests are giving me a tough time, since there is a lot of data that doesn't meet the criteria.

Reply:
Unique failure on ID is strange. Means there might be duplicate data, so other steps need to be taken in order to deduplicate them.

Reply:
Failure on `id` is resolved, it was caused by duplicates from `received_at_date`

Reply:
That's awesome! If `id` check is resolved, let's remove the `combination_of_columns` test. The uniqueness will always be true because an already unique column (`id`) is included in the columns. It is also taking a long time, make it impractical to be run on every PR/hourly.

Reply:
Removed the `combination_of_columns` test.